src/preferences/upgrade.cpp

```diff
@@ -412,11 +410,8 @@ UserSettingsPointer Upgrade::versionUpgrade(const QString& settingsPath) {
             qDebug() << "Upgrade failed!\n";
         }
     }
-
-    if (configVersion.startsWith("1.12") ||
-        configVersion.startsWith("2.0") ||
-        configVersion.startsWith("2.1.0")) {
-        // No special upgrade required, just update the value.
+    else {
+        // Vesrion seems to be never or equal v"1.12" so, no special upgrade required, just update the value.
```

Comment:
Typo and wording

The common else branch is also wrong, because it will silently downgrade to a lower version. I mentioned in my bug report that this case needs to be handled properly. It involves parsing the first 2 or 3 digits (major, minor, patch) from the version number string. You need to consider pre-releases that might have an arbitrary suffix.

Reply:
A Regex should work